otelcol/internal/configunmarshaler/processors.go

```diff
@@ -15,40 +15,37 @@
 package configunmarshaler // import "go.opentelemetry.io/collector/otelcol/internal/configunmarshaler"
 
 import (
+	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/confmap"
-	"go.opentelemetry.io/collector/processor"
 )
 
-// processorsKeyName is the configuration key name for processors section.
-const processorsKeyName = "processors"
+type Configs struct {
+	cfgs map[component.ID]component.Config
 
-type Processors struct {
-	procs map[component.ID]component.Config
-
-	factories map[component.Type]processor.Factory
+	factories map[component.Type]component.Factory
 }
 
-func NewProcessors(factories map[component.Type]processor.Factory) *Processors {
-	return &Processors{factories: factories}
+func NewConfigs(factories map[component.Type]component.Factory) *Configs {
+	return &Configs{factories: factories}
 }
 
-func (p *Processors) Unmarshal(conf *confmap.Conf) error {
+func (p *Configs) Unmarshal(conf *confmap.Conf) error {
 	rawProcs := make(map[component.ID]map[string]interface{})
 	if err := conf.Unmarshal(&rawProcs, confmap.WithErrorUnused()); err != nil {
 		return err
 	}
 
 	// Prepare resulting map.
-	p.procs = make(map[component.ID]component.Config)
+	p.cfgs = make(map[component.ID]component.Config)
 	// Iterate over processors and create a config for each.
 	for id, value := range rawProcs {
 		// Find processor factory based on "type" that we read from config source.
 		factory := p.factories[id.Type()]
 		if factory == nil {
```

Comment:
I tried to use generics with `Configs[F component.Factory]` but I got an error at this line `mismatched types F and untyped nil`. If you have an idea how to fix that, will remove the need of `toFactoryMap` (small func anyway)

Reply:
This should work: https://github.com/open-telemetry/opentelemetry-collector/pull/6801